app/questionnaire/questionnaire_schema.py

```diff
@@ -503,19 +500,9 @@ def _update_answer_dependencies_for_calculated_summary_dependency(
             block_id=dependent_block["id"]
         )
         for answer_id in calculated_summary_answer_ids:
-            if answer_id in [
-                *self._dynamic_answer_ids,
-                *self._repeating_block_answer_ids,
-            ]:
-                # Type ignore: answer_id is valid so block must exist
-                block_id: str = self.get_block_for_answer_id(answer_id)["id"]  # type: ignore
-                if block_id in self._list_dependent_block_additional_dependencies:
-                    for (
-                        list_block_id
-                    ) in self._list_dependent_block_additional_dependencies[block_id]:
-                        self._answer_dependencies_map[list_block_id].add(
-                            answer_dependent
-                        )
+            if list_name := self.get_list_name_for_answer_id(answer_id):
+                # dynamic/repeating answers means the calculated summary also depends on the list those answers loop over
+                self._list_dependencies_map[list_name].add(answer_dependent)
             self._answer_dependencies_map[answer_id].add(answer_dependent)
 
     def _update_answer_dependencies_for_grand_calculated_summary(
```

Comment:
Same here? Do we want to rename it so we mention list dependecies? Should it be general `dependencies`?

Reply:
Yep, good idea, have fixed this in quite a few places